src/content/docs/fr/reference/dev-toolbar-app-reference.mdx

```diff
@@ -0,0 +1,626 @@
+
+#### `on()`
+
+**Signature :** `on<T>(event: string, callback: (data: T) => void) => void`
+<Since v="4.7.0" /><br/>
+
+Enregistre un rappel à appeler lorsque le serveur envoie un message avec l'événement spécifié.
+
+```ts title="src/my-app.js"
+init(canvas, app, server) {
+  server.on('server-message', (data) => {
+    console.log(data.message);
+  });
+}
+```
+
+### Sur le serveur
+
+Dans une intégration, telle que [l'intégration qui ajoute votre application de barre d'outils](#configuration-de-lintégration-des-applications-dans-la-barre-doutils), utilisez le [hook `astro:server:setup`](/fr/reference/integrations-reference/#astroserversetup) pour accéder à l'objet `toolbar` afin d'envoyer et de recevoir des messages vers et à partir de vos applications.
+
+```ts title="my-integration.js" {5}
+export default () => ({
+  name: "my-integration",
+  hooks: {
+    "astro:config:setup": ({ addDevToolbarApp }) => {},
+    "astro:server:setup": ({ toolbar }) => {},
+  },
+});
+```
+
+#### `send()`
+
+**Signature :** `send<T>(event: string, data: T) => void`
+<Since v="4.7.0" /><br/>
+
+Envoie des données au client.
+
+```ts title="my-integration.js"
+'astro:server:setup': ({ toolbar }) => {
+  toolbar.send('server-message', { message: 'Bonjour !' });
+},
+```
+
+#### `on()`
+
+**Signature :** `on<T>(event: string, callback: (data: T) => void) => void`
+<Since v="4.7.0" /><br/>
+
+Enregistre un callback qui sera appelé lorsque le client enverra un message avec l'événement spécifié.
+
+```ts title="my-integration.js"
+'astro:server:setup': ({ toolbar }) => {
+  toolbar.on('my-app:my-message', (data) => {
+    console.log(data.message);
+  });
+},
+```
+
+#### `onInitialized()`
+
+**Signature :** `onInitialized(appId: string, callback: () => void) => void`
+<Since v="4.7.0" /><br/>
+
+Enregistre un callback qui sera appelé lors de l'initialisation de l'application.
+
+```ts title="my-integration.js"
+'astro:server:setup': ({ toolbar }) => {
+  toolbar.onInitialized('my-app', () => {
+    console.log('L'application est maintenant initialisée !');
+  });
+},
+```
+
+:::note
+cL'événement intégré `connection` de Vite se déclenche **avant** que les applications de la barre d'outils Dev ne soient initialisées et ne peut donc pas être utilisé directement par les applications. Utilisez la méthode `onInitialized` pour vous assurer que l'application est complètement initialisée avant de lui envoyer des messages.
+:::
+
+#### `onAppToggled()`
+
+**Signature :** `onAppToggled(appId: string, callback: (options: {state: boolean}) => void) => void`
+<Since v="4.7.0" /><br/>
+
+Enregistre un rappel à appeler lorsque l'utilisateur clique sur l'icône de l'application dans l'interface utilisateur pour activer ou désactiver l'application.
```

Comment:
```suggestion
Enregistre une fonction de rappel à appeler lorsque l'utilisateur clique sur l'icône de l'application dans l'interface utilisateur pour activer ou désactiver l'application.
```
Alternatively, since you already use `callback`:

```suggestion
Enregistre un callback à appeler lorsque l'utilisateur clique sur l'icône de l'application dans l'interface utilisateur pour activer ou désactiver l'application.
```